src/nlsolve/solver_state_results.jl

```diff
@@ -65,16 +65,16 @@ function update!(tr::SolverTrace,
     return
 end
 
-mutable struct SolverResults{T,I<:AbstractArray{T},Z<:AbstractArray{T}}
+mutable struct SolverResults{T<:Number,I<:AbstractArray{T},Z<:AbstractArray{T}}
     method::String
     initial_x::I
     zero::Z
-    residual_norm::T
+    residual_norm::Real
```

Comment:
Do you have an idea how this would be implemented, because one can not simply use `real(T)` there as it will give rise to a `LoadError` (no function matching `real(::TypeVar)`).
And I can not think of a way to use.
The same would be helpful to make `cache.d` a real array of the appropriate length and type.

Reply:
wouldn't it be sufficient to define the struct with an additional type `rT<:Real` ?

Reply:
Like this looks like it works:

```
julia> f(x::T) where {rT<:Real, T<: Union{rT,Complex{rT}}} = rT
f (generic function with 1 method)

julia> f(2)
Int64

julia> f(2im)
Int64

```

Reply:
OK this is strange, now it works for some reason with the same code... :smile: